### PR TITLE
refactor: split Overlay into a new component

### PR DESCRIPTION
**Pull request by EvanBacon**
- Disable touch highlight on mobile Safari

### CODE REVIEW

File: packages/drawer/src/views/Overlay.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+const { interpolate, cond, greaterThan } = Animated;
+
+const PROGRESS_EPSILON = 0.05;
+
+type Props = React.ComponentProps<typeof Animated.View> & {
+  progress: Animated.Node<number>;
+};
+
+const Overlay = React.forwardRef(function Overlay(
```
Comment (satya164) [inline packages/drawer/src/views/Overlay.tsx:13]: Do we need the `forwardRef` wrapper?

Comment (EvanBacon) [reply packages/drawer/src/views/Overlay.tsx:13]: This works better with gestures on web

Comment (satya164) [reply packages/drawer/src/views/Overlay.tsx:13]: You mean gesture handler needs it?

File: packages/drawer/src/index.tsx [hunk 2]
```diff
@@ -11,6 +11,7 @@ export { default as DrawerItem } from './views/DrawerItem';
 export { default as DrawerItemList } from './views/DrawerItemList';
 export { default as DrawerContent } from './views/DrawerContent';
 export { default as DrawerContentScrollView } from './views/DrawerContentScrollView';
+export { default as Overlay } from './views/Overlay';
```
Comment (satya164) [inline packages/drawer/src/index.tsx]: I think we don't need to export it for the public API. Can you remove it? Then I'll merge

```suggestion
```

**Review by satya164 — APPROVED** [review]

**Pull request merged**